generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/internal/extensions/kClassExtensions.kt

```diff
@@ -61,7 +61,7 @@ internal fun KClass<*>.isInterface(): Boolean =
     this.java.isInterface || this.isAbstract || this.isSealed
 
 internal fun KClass<*>.isUnion(): Boolean =
-    this.isInterface() && this.declaredMemberProperties.isEmpty() && this.declaredMemberFunctions.isEmpty()
+    (this.isInterface() && this.declaredMemberProperties.isEmpty() && this.declaredMemberFunctions.isEmpty())  || this.isInstance(Any::class)
```

Comment:
this is a little too loose -> in 99.9% of cases using `Any` as return type should blow up, this should check if it is `Any` and has annotation

Reply:
Working on a better check